zipline/data/hdf5_daily_bars.py

```diff
@@ -466,41 +466,63 @@ def load_raw_arrays(self,
             (minutes in range, sids) with a dtype of float64, containing the
             values for the respective field over start and end dt range.
         """
-        self._validate_assets(assets)
         self._validate_timestamp(start_date)
         self._validate_timestamp(end_date)
 
         start = start_date.asm8
         end = end_date.asm8
 
-        sid_selector = self.sids.searchsorted(assets)
+        sid_selector, out_buf_indexer = self._make_sid_indexers(assets)
+
         date_slice = self._compute_date_range_slice(start, end)
 
-        nrows = date_slice.stop - date_slice.start
-        ncols = len(self.sids)
+        n_dates = date_slice.stop - date_slice.start
+        n_query_sids = len(assets)
```

Comment:
The fact that the order of these two variables is transposed from the order in which they're used to initialize `read_buf`/`out_buf` below tripped me up a bit when trying to keep these variables straight.